ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/PendingRequests.tsx

```diff
@@ -31,23 +37,83 @@ import {
     RequestScope,
 } from './components/ExceptionRequestTableCells';
 import FilterAutocompleteSelect from '../components/FilterAutocomplete';
-import { pendingRequests as vulnerabilityExceptions } from './mockUtils';
+import TableErrorComponent from '../WorkloadCves/components/TableErrorComponent';
 
 const searchOptions: SearchOption[] = [
-    REQUEST_ID_SEARCH_OPTION,
+    REQUEST_NAME_SEARCH_OPTION,
     IMAGE_CVE_SEARCH_OPTION,
     REQUESTER_SEARCH_OPTION,
     IMAGE_SEARCH_OPTION,
 ];
 
+const sortFields = [
+    'Request Name',
+    'Requester User Name',
+    'Created Time',
+    'Request Expiry Time',
+    'Image Registry Scope',
+];
+const defaultSortOption = {
+    field: sortFields[0],
+    direction: 'desc',
+} as const;
+
 function PendingApprovals() {
     const { searchFilter, setSearchFilter } = useURLSearch();
     const { page, perPage, setPage, setPerPage } = useURLPagination(20);
+    const { sortOption, getSortParams } = useURLSort({
+        sortFields,
+        defaultSortOption,
+        onSort: () => setPage(1),
+    });
+
+    const vulnerabilityExceptionsFn = useCallback(
+        () =>
+            fetchVulnerabilityExceptions(
+                {
+                    ...searchFilter,
+                    'Request Status': ['PENDING', 'APPROVED_PENDING_UPDATE'],
+                },
+                sortOption,
+                page - 1,
+                perPage
+            ),
+        [searchFilter, sortOption, page, perPage]
+    );
+    const { data, loading, error } = useRestQuery(vulnerabilityExceptionsFn);
 
     function onFilterChange() {
         setPage(1);
     }
 
+    if (loading && !data) {
+        return (
+            <Bullseye>
+                <Spinner isSVG />
+            </Bullseye>
+        );
+    }
+
+    if (error) {
+        return (
+            <PageSection variant="light">
+                <TableErrorComponent
+                    error={error}
+                    message="An error occurred. Try refreshing again"
+                />
+            </PageSection>
+        );
+    }
+
+    if (!data) {
+        return (
+            <NotFoundMessage
+                title="404: We couldn't find that page"
+                message="Pending vulnerability exception requests could not be found."
+            />
+        );
+    }
```

Comment:
Nit: is this necessary? Is there a situation where `loading === false` and `error === undefined` where `data` will not be a truthy value?

Reply:
At runtime, I don't think so. But Typescript thinks it can still be undefined. The type for `data` is `VulnerabilityException[] | undefined`. I think we can get around this if `useRestQuery` returned `data` of type `ReturnType` rather than `ReturnType | undefined`.